packages/vsce/src/commands/showLogsCommand.ts

```diff
@@ -0,0 +1,122 @@
+  // also filter so we only automatically pick z/os connections that have credentials associated
+  zosProfiles = zosProfiles.sort((prof) => (prof.profile.type === "zosmf" ? -1 : 1)).filter((prof) => prof.profile.user);
+
+  // filter out profiles that are not in the same base as the cics profile
+  const matchingProfiles: IProfileLoaded[] = [];
+  if (baseForCicsProfile) {
+    for (const profile of zosProfiles) {
+      if (baseForCicsProfile && baseForCicsProfile?.name === (await ProfileManagement.getProfilesCache().fetchBaseProfile(profile.name))?.name) {
+        matchingProfiles.push(profile);
+      }
+    }
+  }
+
+  if (matchingProfiles.length > 0) {
+    return matchingProfiles[0];
+  }
+
+  // we couldn't find anything within a profile group
+  // filter down to just profiles that have the same hostname as our cics connection
+  const cicsHostProfiles = zosProfiles.filter((profile) => cicsProfile.profile.host === profile.profile.host);
+
+  return cicsHostProfiles[0];
+}
+
+function promptUserForProfile(zosProfiles: IProfileLoaded[]): Thenable<string> {
+  const profileNames = zosProfiles.map((value) => {
+    return value.name;
+  });
+
+  if (profileNames.length > 0) {
+    const quickPickOptions: vscode.QuickPickOptions = {
+      placeHolder: vscode.l10n.t("Select a profile to access the logs"),
+      ignoreFocusOut: true,
+      canPickMany: false,
+    };
+    return Gui.showQuickPick(profileNames, quickPickOptions);
+  } else {
+    return null;
+  }
+}
+
+export async function getJobIdForRegion(selectedRegion: CICSRegionTree): Promise<string> {
+  // when we have a CICSRGN table we have jobid, but not when we have a
+  // MAS table. we get either CICSRGN or MAS depending on where we are in the
+  // tree. request CICSRGN if jobid isn't available.
+  let jobid: string = selectedRegion.region.jobid;
+  if (!jobid) {
+    let response: ICMCIApiResponse;
+    try {
+      response = await getResource(selectedRegion.parentSession.getSession(), {
+        name: "CICSRegion",
+        cicsPlex: selectedRegion.parentPlex.plexName,
+        regionName: selectedRegion.region.cicsname,
+      });
+      jobid = response.response.records.cicsregion.jobid;
+    } catch (ex) {
+      console.log(ex);
+      // unlikely to get here but logging this would be useful in future
+    }
+  }
+  return jobid;
+}
+
+export function getShowRegionLogs(treeview: TreeView<any>) {
+  return commands.registerCommand("cics-extension-for-zowe.showRegionLogs", async (node) => {
+    const allSelectedRegions = findSelectedNodes(treeview, CICSRegionTree, node);
+    if (!allSelectedRegions || !(allSelectedRegions.length == 1)) {
+      await window.showErrorMessage("One CICS region must be selected");
+      return;
+    }
+    const selectedRegion: CICSRegionTree = allSelectedRegions[0];
+    const zosProfiles = (await ProfileManagement.getProfilesCache().fetchAllProfiles()).filter(
+      (profile) => profile.type === "zosmf" || profile.type === "rse",
```

Comment:
By no means mandatory, but I like doing something like this when checking if a value is 1 of multiple options.

```suggestion
      (profile) => ["zosmf", "rse"].includes(profile.type),
```